python/lsst/ts/mtdome/mtdome_csc.py

```diff
@@ -511,48 +555,64 @@ async def write_then_read_reply(
             TimeoutValue} where "response" can be zero for "OK" or non-zero
             for "ERROR".
         """
-        command_dict = dict(command=command, parameters=params)
-        st = encoding_tools.encode(**command_dict)
+        command_id = next(self._index_iter)
+        self.commands_without_reply[command_id] = CommandTime(
+            command=command, tai=utils.current_tai()
+        )
+        command_dict = dict(commandId=command_id, command=command, parameters=params)
         async with self.communication_lock:
             try:
-                assert self.writer is not None
+                assert self.client is not None
             except AssertionError as e:
-                await self.fault(code=3, report=f"Error writing command {st}: {e}")
+                await self.fault(
```

Comment:
Instead of try/except consider something like this:
```
if self.client is None:
    await self.fault(...)
    raise RuntimeError("No client")
```